common/src/main/kotlin/serialization/DurationSerializers.kt

```diff
@@ -0,0 +1,66 @@
+package dev.kord.common.serialization
+
+import kotlinx.serialization.KSerializer
+import kotlinx.serialization.descriptors.PrimitiveKind
+import kotlinx.serialization.descriptors.PrimitiveSerialDescriptor
+import kotlinx.serialization.descriptors.SerialDescriptor
+import kotlinx.serialization.encoding.Decoder
+import kotlinx.serialization.encoding.Encoder
+import kotlin.time.Duration
+import kotlin.time.DurationUnit
+import kotlin.time.DurationUnit.*
+import kotlin.time.toDuration
+
+
+/** Serializer that encodes and decodes [Duration]s. */
+public sealed class DurationSerializer(private val unit: DurationUnit, name: String) : KSerializer<Duration> {
+
+    final override val descriptor: SerialDescriptor =
+        PrimitiveSerialDescriptor("dev.kord.common.serialization.$name", PrimitiveKind.LONG)
+
+    protected abstract fun Duration.convert(): Long
```

Comment:
Why have this method if toLong(DurationUnit) exists?

Reply:
Fair point, didn't notic tbh. I will replace it then.